lib/ansible/modules/database/influxdb/influxdb_user.py

```diff
@@ -173,7 +173,7 @@ def set_user_grants(module, client, user_name, grants):
                 if v['privilege'] == 'ALL PRIVILEGES':
                     v['privilege'] = 'ALL'
                     current_grants[i] = v
-            parsed_grants.add(current_grants[i])
+                parsed_grants.add(current_grants[i])
```

Comment:
`current_grants[i] = v` is useless (as we don't use current_grants later), just add v to parsed_grants at the end of the loop :
`parsed_grants.add(v)`

Anyway, that's the solution I was going to purpose after hitting the same bug.

Reply:
Noted and updated. Thanks